gammapy/data/data_store.py

```diff
@@ -226,14 +237,19 @@ def obs(self, obs_id):
         kwargs = {"obs_id": int(obs_id)}
         kwargs["obs_info"] = table_row_to_dict(row)
 
-        hdu_list = ["events", "gti", "aeff", "edisp", "psf", "bkg", "rad_max"]
-
-        for hdu in hdu_list:
+        if required_irf == "full-enclosure":
+            hdus = ["events", "gti", "aeff", "edisp", "psf", "bkg"]
```

Comment:
I think it could be good to define a dict at the top of the file, like:

```python3
REQUIRED_HDUS = {
    "full-enclosure": ...,
    "point-like": ...,
    "all": ...,
}
```
To have the definite of those in a central place.